### DISCUSSION TITLE
Add factory methods for libgdx primitive type arrays

### DISCUSSION THREAD

**Discussion by Divelix** [libktx/ktx#333] [Ideas]
I just used `com.badlogic.gdx.utils.FloatArray` to serialize array of floats to json and came up with the fact that I can't generate them easily for no argument constructor, so I did that thing:
```kotlin
fun gdxFloatArrayOf(vararg elements: Float): GdxFloatArray = GdxFloatArray(elements)
``` 
And now I can write concise data classes for serialization purposes:
```kotlin
data class A(
        val floats1: GdxFloatArray = gdxFloatArrayOf(),
        val floats2: GdxFloatArray = gdxFloatArrayOf(),
)
```
I just thought it would be nice to have factory functions for all libgdx primitive type arrays in `ktx.collections`.
Here are the ones I could find:
- [x] `BooleanArray`
- [x] `ByteArray`
- [x] `CharArray`
- [x] `FloatArray`
- [x] `IntArray`
- [x] `LongArray`
- [x] `ShortArray`

**Comment by czyzby** [comment]
Makes sense, thanks for the suggestion. Turns out I cannot convert the discussion into an issue, but I've created a new one to track progress. If you'd like to submit a pull request, please reference #334. Thanks!

**Reply by czyzby** [reply]
The factory methods were implemented by @vrudas in #334. Thanks!
